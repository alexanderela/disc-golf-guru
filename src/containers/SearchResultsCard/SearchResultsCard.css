.SearchResultsCard {
  align-items: center;
  background-color: #FFF;
  border: 3px solid #febb00;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  height: 20rem;
  justify-content: flex-start;
  margin: 1rem auto;
  padding: 0 0 1rem 0;
  width: 38rem;
}

.course-search-result {
  align-items: center;
  display: flex;
  margin: 0 0 1.5rem;
  overflow-wrap: break-word;
}

.search-result-btn {
	background-color: #febb00;
  border: none;
  border-radius: 10px;
  color: #524949;
  cursor: pointer;
  font-weight: 700;
  margin: 0 0 0 0.5rem;   
  outline-width: 0;
}

.search-result-btn:hover {
	background-color: #2c9a2c;
	color: #FFF;
}

.search-result-header {
  color: #524949;
  font-size: 1.5rem;
  font-weight: 600;
  margin: 1rem 0;
}

.nearby-courses {
  color: #2c9a2c;
  font-size: 1.25rem;
  font-weight: 600;
  margin: 0 0 1rem 0;
}