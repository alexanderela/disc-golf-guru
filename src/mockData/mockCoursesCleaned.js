const mockCoursesCleaned = [
  {
    id: '4712',
    name: 'Genesee Valley Park',
    isFavorite: false,
    holes: '18',
    city: 'Rochester',
    state: 'NY',
    country: 'United States',
    zip: '14620',
    address: '99 Elmwood Ave.',
    rating: '3.29',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
  {
    id: '7628',
    name: 'Lookup Park',
    isFavorite: false,
    holes: '9',
    city: 'Henrietta',
    state: 'NY',
    country: 'United States',
    zip: '14467',
    address: '3850 East Henrietta Rd.',
    rating: '3.13',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
  {
    id: '4937',
    name: 'Monroe Community College DGC',
    isFavorite: false,
    holes: '18',
    city: 'Henrietta',
    state: 'NY',
    country: 'United States',
    zip: '14623',
    address: '',
    rating: '2.94',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
  {
    id: '6795',
    name: 'Ellison Park',
    isFavorite: false,
    holes: '18',
    city: 'Rochester',
    state: 'NY',
    country: 'United States',
    zip: '14610',
    address: '',
    rating: '3.54',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
  {
    id: '6443',
    name: 'North Ponds DGC',
    isFavorite: false,
    holes: '9',
    city: 'Webster',
    state: 'NY',
    country: 'United States',
    zip: '14580',
    address: '750 Holt Rd.',
    rating: '1.75',
    isPayToPlay: 'No',
    isPrivate: 'No',
  },
];

export default mockCoursesCleaned;
