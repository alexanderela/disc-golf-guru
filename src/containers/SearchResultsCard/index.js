import React from 'react';
import './SearchResultsCard.css'
import { withRouter, Link } from 'react-router-dom';

const SearchResultsCard = (props) => {
	const courseResults = props.courses.map(course => {
		const { name, address, city, state, zip, id } = course
		return <div className='course-search-result' key={name}>
							<p>{`${name} ${address}, ${city}, ${state} ${zip}`}</p>
							<Link to={`/findcourses/searchresults/courseinfo/${id}`}>
								<button 
									className='search-result-btn'>Select
								</button>
							</Link>
						</div>
	})
	return(
		<div className='SearchResultsCard'>
			
				<h2 className='search-result-header'>Search Results</h2>
			<h4 className='nearby-courses'>Nearby Courses</h4>
			{props.courses.length ? courseResults : 'Oops! There are no courses near where you searched.'}
		</div>
	)
}

export default withRouter(SearchResultsCard)