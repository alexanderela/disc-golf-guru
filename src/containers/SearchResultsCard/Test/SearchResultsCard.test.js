import React from 'react';
import ReactDOM from 'react-dom';
import SearchResultsCard from '../';
import { shallow } from 'enzyme';
import mockCoursesCleaned from '../../../mockData/mockCoursesCleaned.js';
import { mapDispatchToProps } from '../';

describe('SearchResultsCard', () => {
  let wrapper;
  let mockCourses;

  beforeEach(() => {
    wrapper = shallow(<SearchResultsCard courses={mockCoursesCleaned} />);
  });

  it('should render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

	describe('mapDispatchToProps', () => {
    let dispatch;

    beforeEach(() => {
      dispatch = jest.fn();
    });

    it('should map a key of showSearchResultsCard', () => {
      const dispatchedProps = mapDispatchToProps(dispatch);
      expect(dispatchedProps.showSearchResultsCard).toBeDefined();
    });

    it('should map a key of showCourseDetailsCard', () => {
      const dispatchedProps = mapDispatchToProps(dispatch);
      expect(dispatchedProps.showCourseDetailsCard).toBeDefined();
    });

    it('should have showSearchResultsCard call dispatch', () => {
      const dispatchedProps = mapDispatchToProps(dispatch);
      dispatchedProps.showSearchResultsCard(mockCoursesCleaned);
      expect(dispatch).toHaveBeenCalled();
    });

    it('should have showCourseDetailsCard call dispatch', () => {
      const dispatchProps = mapDispatchToProps(dispatch);
      dispatchProps.showCourseDetailsCard();
      expect(dispatch).toHaveBeenCalled();
    });
  });
});
