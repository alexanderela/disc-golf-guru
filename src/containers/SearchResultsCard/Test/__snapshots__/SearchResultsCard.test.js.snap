// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsCard should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Connect(SearchResultsCard))
    courses={
      Array [
        Object {
          "address": "99 Elmwood Ave.",
          "city": "Rochester",
          "country": "United States",
          "holes": "18",
          "id": "4712",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "Genesee Valley Park",
          "rating": "3.29",
          "state": "NY",
          "zip": "14620",
        },
        Object {
          "address": "3850 East Henrietta Rd.",
          "city": "Henrietta",
          "country": "United States",
          "holes": "9",
          "id": "7628",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "Lookup Park",
          "rating": "3.13",
          "state": "NY",
          "zip": "14467",
        },
        Object {
          "address": "",
          "city": "Henrietta",
          "country": "United States",
          "holes": "18",
          "id": "4937",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "Monroe Community College DGC",
          "rating": "2.94",
          "state": "NY",
          "zip": "14623",
        },
        Object {
          "address": "",
          "city": "Rochester",
          "country": "United States",
          "holes": "18",
          "id": "6795",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "Ellison Park",
          "rating": "3.54",
          "state": "NY",
          "zip": "14610",
        },
        Object {
          "address": "750 Holt Rd.",
          "city": "Webster",
          "country": "United States",
          "holes": "9",
          "id": "6443",
          "isFavorite": false,
          "isPayToPlay": "No",
          "isPrivate": "No",
          "name": "North Ponds DGC",
          "rating": "1.75",
          "state": "NY",
          "zip": "14580",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
